app/connectors/httpParsers/GetSubmittedInterestIfHttpParser.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors.httpParsers
+
+import models.{ErrorModel, InterestDetailsIfModel}
+import play.api.Logging
+import play.api.http.Status._
+import uk.gov.hmrc.http.{HttpReads, HttpResponse}
+import utils.PagerDutyHelper.PagerDutyKeys._
+import utils.PagerDutyHelper.pagerDutyLog
+
+object GetSubmittedInterestIfHttpParser extends APIParser with Logging {
+  type SubmittedInterestIfResponse = Either[ErrorModel, InterestDetailsIfModel]
```

Comment:
You've introduced a new model named `InterestDetailsIfModel`. You shouldn't need to introduce a new model for this.

You can add the optional `deletedPeriod` field to the existing `InterestDetailsModel` with a default value of `None`.